commit 3cce0650e415f7b7713b1ebf1f2718cd9bb634b2
Author: Roman Meshkevich <romik.momik@trikset.com>
Date:   Tue Aug 13 13:52:59 2013 +0400

    booting firmware fix

diff --git a/src/udevd.c b/src/udevd.c
index 1702217..e3af967 100644
--- a/src/udevd.c
+++ b/src/udevd.c
@@ -104,6 +104,7 @@ struct event {
         const char *devpath_old;
         dev_t devnum;
         bool is_block;
+	bool is_firmware;
         int ifindex;
 };
 
@@ -453,6 +454,7 @@ static int event_queue_insert(struct udev_device *dev)
         event->devpath_old = udev_device_get_devpath_old(dev);
         event->devnum = udev_device_get_devnum(dev);
         event->is_block = (strcmp("block", udev_device_get_subsystem(dev)) == 0);
+	event->is_firmware = (strcmp("firmware",udev_device_get_subsystem(dev)) == 0);
         event->ifindex = udev_device_get_ifindex(dev);
 
         udev_queue_export_device_queued(udev_queue_export, dev);
@@ -543,7 +545,7 @@ static bool is_devpath_busy(struct event *event)
                 }
 
                 /* parent device event found */
-                if (event->devpath[common] == '/') {
+                if (event->devpath[common] == '/'  && !event->is_firmware ) {
                         event->delaying_seqnum = loop_event->seqnum;
                         return true;
                 }
