diff -ruN sphinxbase-0.8.origin/test/regression/Makefile.am sphinxbase-0.8/test/regression/Makefile.am
--- sphinxbase-0.8.origin/test/regression/Makefile.am	2013-10-17 22:07:45.616114850 +0400
+++ sphinxbase-0.8/test/regression/Makefile.am	2013-10-17 22:08:20.872115722 +0400
@@ -33,9 +33,9 @@
 
 # Disable sphinx_fe tests for now if fixed-point due to imprecision
 if FIXED_POINT
-TESTS = $(srcdir)/test-cepview.sh $(srcdir)/test-sphinx_pitch.sh
+TESTS = test-cepview.sh test-sphinx_pitch.sh
 else
-TESTS = $(srcdir)/test-*.sh
+TESTS = test-*.sh
 endif
 
 EXTRA_DIST += $(TESTS)
