diff -rupN fbgrab.old/fbgrab.c fbgrab/fbgrab.c
--- fbgrab.old/fbgrab.c	2014-11-10 18:50:09.280928269 +0300
+++ fbgrab/fbgrab.c	2014-11-10 18:50:56.648928616 +0300
@@ -169,11 +169,12 @@ static void get_framebufferdata(char *de
         fprintf(stderr, "alpha: offset: %i, length: %i, msb_right: %i\n", fb_varinfo_p->transp.offset, fb_varinfo_p->transp.length, fb_varinfo_p->transp.msb_right);
         fprintf(stderr, "pixel format: %s\n", fb_varinfo_p->nonstd == 0 ? "standard" : "non-standard");
     }
+#if 0
     Blue = fb_varinfo_p->blue.offset >> 3;
     Green = fb_varinfo_p->green.offset >> 3;
     Red = fb_varinfo_p->red.offset >> 3;
     Alpha = fb_varinfo_p->transp.offset >> 3;
-
+#endif
     (void) close(fd);
 }
 
