diff -rupN fbgrab.old/fbgrab.c fbgrab/fbgrab.c
--- fbgrab.old/fbgrab.c	2015-10-15 17:35:36.303659072 +0300
+++ fbgrab/fbgrab.c	2015-10-15 17:53:32.983685751 +0300
@@ -44,7 +44,7 @@ static int Alpha = 3;
 
 static void usage(char *binary)
 {
-    fprintf(stderr, "Usage:   %s\t[-hi] [-{C|c} vt] [-d dev] [-s n] [-z n]\n"
+    fprintf(stderr, "Usage:   %s\t[-hi] [-{C|c} vt] [-d dev] [-S] [-s n] [-z n]\n"
 	   "\t\t[-f fromfile -w n -h n -b n] filename.png\n", binary);
 }
 
@@ -64,6 +64,7 @@ static void help(char *binary)
     fprintf(stderr, "\t-h n  \tset height to n pixels, required when reading from file\n"
 	   "\t\tcan be used to force height when reading from framebuffer\n");
     fprintf(stderr, "\t-i    \tturns on interlacing in PNG\n");
+    fprintf(stderr, "\t-S    \tswap blue and red component(TRIK display)\n");
     fprintf(stderr, "\t-s n  \tsleep n seconds before making screenshot\n");
     fprintf(stderr, "\t-v    \tverbose, print debug information.\n");
     fprintf(stderr, "\t-w n  \tset width to n pixels, required when reading from file\n"
@@ -169,6 +170,7 @@ static void get_framebufferdata(char *de
         fprintf(stderr, "alpha: offset: %i, length: %i, msb_right: %i\n", fb_varinfo_p->transp.offset, fb_varinfo_p->transp.length, fb_varinfo_p->transp.msb_right);
         fprintf(stderr, "pixel format: %s\n", fb_varinfo_p->nonstd == 0 ? "standard" : "non-standard");
     }
+
 #if 0
     Blue = fb_varinfo_p->blue.offset >> 3;
     Green = fb_varinfo_p->green.offset >> 3;
@@ -404,7 +406,7 @@ int main(int argc, char **argv)
 
     for(;;)
     {
-	optc=getopt(argc, argv, "f:z:w:b:h:iC:c:d:s:?v");
+	optc=getopt(argc, argv, "f:z:w:b:h:iC:c:d:s:?vS");
 	if (optc==-1)
 	    break;
 	switch (optc) 
@@ -437,6 +439,11 @@ int main(int argc, char **argv)
 	case 'v':
 	    verbose = 1;
 	    break;
+	case 'S':
+		Red = Red^Blue;
+		Blue= Red^Blue;
+		Red = Blue^Red;
+		break;
 	case 's':
 	    (void) sleep((unsigned int) atoi(optarg));
 	    break;
