diff -rupN systec_can/Makefile systec_can.new/Makefile
--- systec_can/Makefile	2015-03-10 18:41:10.000000000 +0300
+++ systec_can.new/Makefile	2015-07-06 14:54:25.381297501 +0300
@@ -1,48 +1,23 @@
-
-# ------------------------- Secondary Call ----------------------------------
-ifneq ($(KERNELRELEASE),)
-
-#EXTRA_CFLAGS = $(INCLUDES) $(MODFLAGS)
-EXTRA_CFLAGS = -g
-
 obj-m :=	systec_can.o
-
-
-# ------------------------- Primary Call ------------------------------------
-else
-
-.EXPORT_ALL_VARIABLES:
-KDIR			:= /lib/modules/$(shell uname -r)/build
-PWD			:= $(shell pwd)
+SRC			:= $(shell pwd)
 FW_DIR			:= /lib/firmware
 FIRMWARE_FILES		:= systec_can-*.fw
 
 # ------------- Default-Target -------------
 all:
-			$(MAKE) -C $(KDIR) M=$(PWD) modules
+			$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules
 
 modules_install:
-			$(MAKE) -C $(KDIR) M=$(PWD) modules_install
-			@[ -r $(KDIR)/System.map ] || depmod -a
+			$(MAKE) -C $(KERNEL_SRC) M=$(SRC) modules_install
 
 firmware_install:
-			@mkdir -p $(FW_DIR)
-			@install --mode=644 $(FIRMWARE_FILES) $(FW_DIR)
+			@mkdir -p $(INSTALL_MOD_PATH)/$(FW_DIR)
+			@install --mode=644 $(SRC)/$(FIRMWARE_FILES) $(INSTALL_MOD_PATH)/$(FW_DIR)
 			@echo "Firmware files installed to $(FW_DIR)"
 
 install:		modules_install firmware_install
 
-firmware:
-			$(MAKE) -C hex2bin firmware
-
-dist:
-			git archive --format=tar --prefix=systec_can/ $(shell git describe) | bzip2 > systec_can-$(shell git describe).tar.bz2
-
-endif
-
-
-
 clean:
-			$(MAKE)	-C $(KDIR) M=$(PWD) clean
+			$(MAKE)	-C $(KERNEL_SRC) M=$(SRC) clean
 
 .PHONY: all clean
