diff -rupN mjpg-streamer.orig/Makefile mjpg-streamer/Makefile
--- mjpg-streamer.orig/Makefile	2014-07-10 10:48:20.607107285 +0400
+++ mjpg-streamer/Makefile	2014-07-10 10:50:38.875110712 +0400
@@ -13,7 +13,6 @@
 DESTDIR = /usr/local
 
 # set the compiler to use
-CC = gcc
 
 SVNDEV := -D'SVN_REV="$(shell svnversion -c .)"'
 CFLAGS += $(SVNDEV)
diff -rupN mjpg-streamer.orig/plugins/input_control/Makefile mjpg-streamer/plugins/input_control/Makefile
--- mjpg-streamer.orig/plugins/input_control/Makefile	2014-07-10 10:48:20.587107284 +0400
+++ mjpg-streamer/plugins/input_control/Makefile	2014-07-10 10:49:45.655109393 +0400
@@ -7,8 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
-
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
 CFLAGS += -O2 -DLINUX -D_GNU_SOURCE -Wall -shared -fPIC
diff -rupN mjpg-streamer.orig/plugins/input_file/Makefile mjpg-streamer/plugins/input_file/Makefile
--- mjpg-streamer.orig/plugins/input_file/Makefile	2014-07-10 10:48:20.567107284 +0400
+++ mjpg-streamer/plugins/input_file/Makefile	2014-07-10 10:49:39.807109247 +0400
@@ -7,7 +7,7 @@
 #
 ###############################################################
 
-CC = gcc
+
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/input_gspcav1/Makefile mjpg-streamer/plugins/input_gspcav1/Makefile
--- mjpg-streamer.orig/plugins/input_gspcav1/Makefile	2014-07-10 10:48:20.579107284 +0400
+++ mjpg-streamer/plugins/input_gspcav1/Makefile	2014-07-10 10:49:35.939109152 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/input_testpicture/Makefile mjpg-streamer/plugins/input_testpicture/Makefile
--- mjpg-streamer.orig/plugins/input_testpicture/Makefile	2014-07-10 10:48:20.575107284 +0400
+++ mjpg-streamer/plugins/input_testpicture/Makefile	2014-07-10 10:49:31.323109037 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/input_uvc/Makefile mjpg-streamer/plugins/input_uvc/Makefile
--- mjpg-streamer.orig/plugins/input_uvc/Makefile	2014-07-10 10:48:20.567107284 +0400
+++ mjpg-streamer/plugins/input_uvc/Makefile	2014-07-10 10:49:25.619108896 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/output_autofocus/Makefile mjpg-streamer/plugins/output_autofocus/Makefile
--- mjpg-streamer.orig/plugins/output_autofocus/Makefile	2014-07-10 10:48:20.563107284 +0400
+++ mjpg-streamer/plugins/output_autofocus/Makefile	2014-07-10 10:49:20.635108771 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/output_file/Makefile mjpg-streamer/plugins/output_file/Makefile
--- mjpg-streamer.orig/plugins/output_file/Makefile	2014-07-10 10:48:20.583107284 +0400
+++ mjpg-streamer/plugins/output_file/Makefile	2014-07-10 10:49:16.547108671 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/output_http/Makefile mjpg-streamer/plugins/output_http/Makefile
--- mjpg-streamer.orig/plugins/output_http/Makefile	2014-07-10 10:48:20.563107284 +0400
+++ mjpg-streamer/plugins/output_http/Makefile	2014-07-10 10:49:08.727108476 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
diff -rupN mjpg-streamer.orig/plugins/output_rtsp/Makefile mjpg-streamer/plugins/output_rtsp/Makefile
--- mjpg-streamer.orig/plugins/output_rtsp/Makefile	2014-07-10 10:48:20.571107284 +0400
+++ mjpg-streamer/plugins/output_rtsp/Makefile	2014-07-10 10:49:09.783108504 +0400
@@ -9,8 +9,6 @@
 
 # Modified to work with the output_udp plugin by D.Zachariadis
 
-CC = gcc
-
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
 CFLAGS += -O2 -DLINUX -D_GNU_SOURCE -Wall -shared -fPIC
diff -rupN mjpg-streamer.orig/plugins/output_udp/Makefile mjpg-streamer/plugins/output_udp/Makefile
--- mjpg-streamer.orig/plugins/output_udp/Makefile	2014-07-10 10:48:20.571107284 +0400
+++ mjpg-streamer/plugins/output_udp/Makefile	2014-07-10 10:49:10.623108524 +0400
@@ -9,8 +9,6 @@
 
 # Modified to work with the output_udp plugin by D.Zachariadis
 
-CC = gcc
-
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
 CFLAGS += -O2 -DLINUX -D_GNU_SOURCE -Wall -shared -fPIC
diff -rupN mjpg-streamer.orig/plugins/output_viewer/Makefile mjpg-streamer/plugins/output_viewer/Makefile
--- mjpg-streamer.orig/plugins/output_viewer/Makefile	2014-07-10 10:48:20.571107284 +0400
+++ mjpg-streamer/plugins/output_viewer/Makefile	2014-07-10 10:49:51.903109546 +0400
@@ -7,7 +7,6 @@
 #
 ###############################################################
 
-CC = gcc
 
 OTHER_HEADERS = ../../mjpg_streamer.h ../../utils.h ../output.h ../input.h
 
