diff -rupN mc-4.8.12.origin/misc/syntax/Syntax.in mc-4.8.12/misc/syntax/Syntax.in
--- mc-4.8.12.origin/misc/syntax/Syntax.in	2014-07-14 18:03:49.676442111 +0400
+++ mc-4.8.12/misc/syntax/Syntax.in	2014-07-14 18:16:00.136434624 +0400
@@ -256,5 +256,8 @@ include jal.syntax
 file ..\*\\.([lL][kK][rR])$ PIC\slinker\sscript\sfile
 include lkr.syntax
 
+file ..\*\\.(qts)$ QtScript\sProgram
+include js.syntax
+
 file .\* unknown
 include unknown.syntax
